lib/handler/status/errors.c

```diff
@@ -0,0 +1,113 @@
+                                          " \"http1-errors-503\": %llu,\n"
+                                          " \"http1-errors-4xx-others\": %llu,\n"
+                                          " \"http1-errors-5xx-others\": %llu,\n"
+                                          " \"http1-errors-others\": %llu,\n"
+                                          " \"http2-errors-protocol\": %llu, \n"
+                                          " \"http2-errors-internal\": %llu, \n"
+                                          " \"http2-errors-flow_control\": %llu, \n"
+                                          " \"http2-errors-settings_timeout\": %llu, \n"
+                                          " \"http2-errors-stream_closed\": %llu, \n"
+                                          " \"http2-errors-frame_size\": %llu, \n"
+                                          " \"http2-errors-refused_stream\": %llu, \n"
+                                          " \"http2-errors-cancel\": %llu, \n"
+                                          " \"http2-errors-compression\": %llu, \n"
+                                          " \"http2-errors-connect\": %llu, \n"
+                                          " \"http2-errors-enhance_your_calm\": %llu, \n"
+                                          " \"http2-errors-inadequate_security\": %llu, \n"
+                                          " \"http2-errors-other\": %llu",
+                                          esc->agg_errors[E_HTTP_400], esc->agg_errors[E_HTTP_403], esc->agg_errors[E_HTTP_404],
+                                          esc->agg_errors[E_HTTP_405], esc->agg_errors[E_HTTP_416], esc->agg_errors[E_HTTP_417],
+                                          esc->agg_errors[E_HTTP_500], esc->agg_errors[E_HTTP_502], esc->agg_errors[E_HTTP_503],
+                                          esc->agg_errors[E_HTTP_4XX], esc->agg_errors[E_HTTP_5XX], esc->agg_errors[E_HTTP_XXX],
+                                          esc->agg_errors[E_HTTP2_PROTOCOL], esc->agg_errors[E_HTTP2_INTERNAL], esc->agg_errors[E_HTTP2_FLOW_CONTROL],
+                                          esc->agg_errors[E_HTTP2_SETTINGS_TIMEOUT], esc->agg_errors[E_HTTP2_STREAM_CLOSED], esc->agg_errors[E_HTTP2_FRAME_SIZE],
+                                          esc->agg_errors[E_HTTP2_REFUSED_STREAM], esc->agg_errors[E_HTTP2_CANCEL], esc->agg_errors[E_HTTP2_COMPRESSION],
+                                          esc->agg_errors[E_HTTP2_CONNECT], esc->agg_errors[E_HTTP2_ENHANCE_YOUR_CALM], esc->agg_errors[E_HTTP2_INADEQUATE_SECURITY],
+                                          esc->agg_errors[E_HTTP2_OTHER]);
+    return ret;
+#undef BUFSIZE
+}
+
+static void errors_status_done(void *priv)
+{
+    struct errors_status_ctx *esc = priv;
+    if (!esc->pool) {
+        free(esc);
+    }
+}
+
+h2o_status_handler_t errors_status_handler = {
+    .type = H2O_STATUS_HANDLER_PER_THREAD,
```

Comment:
Please refrain from using GCC extensions, since it will make porting H2O difficult.